packages/api-v2/src/AuthProvider.tsx

```diff
@@ -23,16 +23,24 @@ type LoginToken = {
     token: string;
 };
 
-// Create the context
-const AuthContext = createContext<AuthContextType | undefined>(undefined);
-
 type AuthProviderProps = {
     children: React.ReactNode;
     cookieTimeout?: number;
     loginIDKey?: string;
+    selectDefaultAccount?: (loginids: NonNullable<ReturnType<typeof getAccountsFromLocalStorage>>) => string;
 };
 
-function waitForLoginAndTokenWithTimeout(loginIDKey?: string, cookieTimeout = 10000) {
+type AuthProviderPropsWithoutChildren = Omit<AuthProviderProps, 'children'>;
+
+// Create the context
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+function waitForLoginAndTokenWithTimeout(
```

Comment:
minor: typically its better to just use primitives (number, string, boolean) as params... so its immediately understandable for readers without having to learn specific type first (in this case, to understand the function definitiokn, I need to understnad AuthProviderPropsWithoutChildren first, then I need to understand AuthProviderProps before I understand this specific function declaration, while with primiteves it would be immediately understandable)
keeping it simple is more important than deduplication (as too much DRY leads to excessive abstraction and unnecesary tight coupling)

Reply:
Yeah, the reason why I created `AuthProviderPropsWithoutChildren` is to avoid typing `selectDefaultAccount` again inside `waitForLoginAndTokenWithTimeout`. I'll revert back to primitives if that is the better approach here 👍